x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_form.tsx

```diff
@@ -0,0 +1,289 @@
+                    defaultMessage: 'Processors JSON editor',
+                  }
+                ),
+              },
+            }}
+          />
+        </FormRow>
+
+        {/* On-failure field */}
+        <FormRow
+          title={
+            <FormattedMessage
+              id="xpack.ingestPipelines.form.onFailureTitle"
+              defaultMessage="Failure processors"
+            />
+          }
+          description={
+            <>
+              <FormattedMessage
+                id="xpack.ingestPipelines.form.onFailureDescription"
+                defaultMessage="Processors to be executed following a failed processor. {learnMoreLink}"
+                values={{
+                  learnMoreLink: (
+                    <EuiLink href={services.documentation.getHandlingFailureUrl()} target="_blank">
+                      {i18n.translate('xpack.ingestPipelines.form.onFailureDocumentionLink', {
+                        defaultMessage: 'Learn more.',
+                      })}
+                    </EuiLink>
+                  ),
+                }}
+              />
+              <EuiSpacer size="m" />
+              <EuiSwitch
+                label={
+                  <FormattedMessage
+                    id="xpack.ingestPipelines.form.onFailureToggleDescription"
+                    defaultMessage="Add on-failure processors"
+                  />
+                }
+                checked={isOnFailureEditorVisible}
+                onChange={e => setIsOnFailureEditorVisible(e.target.checked)}
+                data-test-subj="onFailureToggle"
+              />
+            </>
+          }
+        >
+          {isOnFailureEditorVisible ? (
+            <UseField
+              path="onFailure"
+              component={JsonEditorField}
+              componentProps={{
+                ['data-test-subj']: 'onFailureEditor',
+                euiCodeEditorProps: {
+                  height: '300px',
+                  'aria-label': i18n.translate(
+                    'xpack.ingestPipelines.form.onFailureFieldAriaLabel',
+                    {
+                      defaultMessage: 'On-failure processors JSON editor',
+                    }
+                  ),
+                },
+              }}
+            />
+          ) : (
+            // <FormRow/> requires children or a field
+            // For now, we return an empty <div> if the editor is not visible
+            <div />
+          )}
+        </FormRow>
+
+        <EuiSpacer size="l" />
+
+        {/* Form submission */}
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup>
+              <EuiFlexItem grow={false}>
+                <EuiButton
+                  fill
+                  color="secondary"
+                  iconType="check"
+                  onClick={form.submit}
+                  data-test-subj="submitButton"
+                  disabled={form.isValid === false}
```

Comment:
Here you might need `form.isSubmitted && form.isValid === false`? Haven't tested locally but this way the button does not change state multiple time meanwhile the user fills the form.